diff --git a/src/heap.c b/src/heap.c
index f6f23549..43fe62a0 100644
--- a/src/heap.c
+++ b/src/heap.c
@@ -59,7 +59,10 @@ static bool mi_heap_page_is_valid(mi_heap_t* heap, mi_page_queue_t* pq, mi_page_
   MI_UNUSED(pq);
   mi_assert_internal(mi_page_heap(page) == heap);
   mi_segment_t* segment = _mi_page_segment(page);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-overflow"
   mi_assert_internal(segment->thread_id == heap->thread_id);
+#pragma GCC diagnostic pop
   mi_assert_expensive(_mi_page_is_valid(page));
   return true;
 }
